src/Elastic.CommonSchema.Serilog/LogEventConverter.cs

```diff
@@ -85,55 +103,167 @@ public static Base ConvertToEcs(LogEvent logEvent, IEcsTextFormatterConfiguratio
+			var server = configuration.MapHttpAdapter?.Server?? new Server();
+			server.User = e.Properties.TryGetValue(SpecialKeys.EnvironmentUserName, out var environmentUserName)
 				? new User { Name = environmentUserName.ToString() }
 				: null;
 
-			var hasHost = e.Properties.TryGetValue("Host", out var host);
+			var hasHost = e.Properties.TryGetValue(SpecialKeys.Host, out var host);
 			server.Address = hasHost ? host.ToString() : null;
 			server.Ip = hasHost ? host.ToString() : null;
 			return server;
 		}
 
-		private static Process GetProcess(Thread currentThread)
+		private static Process GetProcess(LogEvent e, bool mapFromCurrentThread)
 		{
-			if (currentThread == null)
+			LogEventPropertyValue processNameProp = null;
+			LogEventPropertyValue processIdProp = null;
+			LogEventPropertyValue threadIdProp = null;
+			e.Properties.TryGetValue(SpecialKeys.ProcessName, out processNameProp);
+			e.Properties.TryGetValue(SpecialKeys.ProcessId, out processIdProp);
+			e.Properties.TryGetValue(SpecialKeys.ThreadId, out threadIdProp);
+			if (processNameProp == null
+				&& processIdProp == null
+				&& threadIdProp == null
+				&& !mapFromCurrentThread)
 				return null;
 
+			var processName = processNameProp?.ToString();
+			var processId = processIdProp?.ToString();
+			var threadId = threadIdProp?.ToString();
+			var pid = int.TryParse(processId ?? "", out var p)
+				? p
+				: (int?)null;
+
+			if (!mapFromCurrentThread)
+			{
+				return new Process
+				{
+					Title = processName,
+					Name = processName,
+					Pid = pid,
+					Thread = int.TryParse(threadId ?? processId ?? "", out var id)
+						? new ProcessThread() { Id = id }
+						: null,
+				};
+			}
+
+			var currentThread = Thread.CurrentThread;
+			var process = TryGetProcess(pid);
+
 			return new Process
 			{
-				Title = currentThread.Name,
-				Name = currentThread.Name,
-				Executable = currentThread.ExecutionContext.GetType().ToString(),
+				Title = process?.MainWindowTitle,
+				Name = process?.ProcessName ?? processName,
+				Pid = process?.Id ?? pid,
+				Executable = process?.ProcessName ?? processName,
 				Thread = new ProcessThread { Id = currentThread.ManagedThreadId }
 			};
 		}
 
+		private static System.Diagnostics.Process TryGetProcess(int? processId)
+		{
+			try
+			{
+				var pid = processId != null
+					? System.Diagnostics.Process.GetProcessById(processId.Value)
+					: System.Diagnostics.Process.GetCurrentProcess();
+				return pid;
+			}
+			catch (Exception)
+			{
+				return null;
+			}
+		}
+
 		private static Log GetLog(LogEvent e, IReadOnlyList<Exception> exceptions, IEcsTextFormatterConfiguration configuration)
 		{
-			var log = new Log { Level = e.Level.ToString("F"), Logger = "Elastic.CommonSchema.Serilog" };
+			var source = e.Properties.TryGetValue(SpecialKeys.SourceContext, out var context)
+				 ? context.ToString()
+				 : SpecialKeys.DefaultLogger ;
+
+			var log = new Log { Level = e.Level.ToString("F"), Logger = source};
```

Comment:
IIRC I believe we are also rendering out the numeric equivalent in the json?

We'd really want to run queries where log.level >= 1 (for example)

Reply:
Ahh we are only emitting the `log.level` as string, mind need an additional field in ECS?

Reply:
Its a very good question, I raised as an issue in the repo - https://github.com/elastic/ecs/issues/683

Reply:
Opened: https://github.com/elastic/ecs-dotnet/pull/11